### PR TITLE
Final merges to TMI

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed
* `ROOMSTATE` have a spitted classes for ChannelState
* `NOTICE` will be raw. Cause there is notice when would be added for
  - Hosted by Channel
  - Setted time for Followers Only. `ROOMSTATE` returns `followers-only` tag as number in Minutes (not seconds). There is `TimeUnit` class when will be converted to seconds.

### Additional notes

Check some added comments. Otherwise delete it, if is necessary. If you want to add some features mentioned #17 i can put it into this PR. Just say where should I put it?

### CODE REVIEW

**Comment by PhilippHeuer** [comment]
Thanks for the PR.

I got this Exception quite a few times when i tested it:
```java
m.p.t.e.EventDispatcher - Unhandled exception caught dispatching event IRCMessageEvent
	at sun.reflect.GeneratedMethodAccessor30.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at me.philippheuer.twitch4j.events.EventDispatcher.lambda$null$5(EventDispatcher.java:246)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.philippheuer.twitch4j.events.EventDispatcher.lambda$null$6(EventDispatcher.java:243)
	at java.util.concurrent.ConcurrentHashMap.forEach(ConcurrentHashMap.java:1597)
	at me.philippheuer.twitch4j.events.EventDispatcher.lambda$null$7(EventDispatcher.java:242)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.concurrent.ConcurrentHashMap$EntrySpliterator.forEachRemaining(ConcurrentHashMap.java:3606)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at me.philippheuer.twitch4j.events.EventDispatcher.lambda$dispatch$11(EventDispatcher.java:241)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at sun.util.locale.LocaleUtils.toLowerString(LocaleUtils.java:89)
	at sun.util.locale.LanguageTag.parse(LanguageTag.java:191)
	at java.util.Locale.forLanguageTag(Locale.java:1568)
	at me.philippheuer.twitch4j.message.irc.listener.IRCEventListener.lambda$onChannelState$0(IRCEventListener.java:247)
	at java.util.HashMap.forEach(HashMap.java:1288)
	at java.util.Collections$UnmodifiableMap.forEach(Collections.java:1505)
	at me.philippheuer.twitch4j.message.irc.listener.IRCEventListener.onChannelState(IRCEventListener.java:244)
	... 24 more
```
Can you take another look at this?

And please open a new PR for the changes in Issue #17.

File: src/main/java/me/philippheuer/twitch4j/message/irc/listener/IRCEventListener.java [hunk 1]
```diff
@@ -213,23 +212,61 @@ public void onChannnelClientLeaveEvent(IRCMessageEvent event) {
...
+	public void onChannelState(IRCMessageEvent event) {
+		if(event.getCommandType().equals("ROOMSTATE")) {
+			// getting Status on channel
+			Channel channel = event.getClient().getChannelEndpoint(event.getChannelId()).getChannel();
+			Map<ChannelStateEvent.ChannelState, Object> states = new HashMap<ChannelStateEvent.ChannelState, Object>();
+			if (event.getTags().size() > 2) {
+				event.getTags().forEach((k, v) -> {
+					switch (k) {
+						case "broadcaster-lang":
+							states.put(ChannelStateEvent.ChannelState.BROADCAST_LANG, Locale.forLanguageTag(v));
```
Comment (stachu540) [inline src/main/java/me/philippheuer/twitch4j/message/irc/listener/IRCEventListener.java]: That's the issue for [this](https://github.com/PhilippHeuer/twitch4j/pull/20#issuecomment-330082243).

**Comment by stachu540** [comment]
Done!

**Comment by PhilippHeuer** [comment]
Looks good now.

**Pull request merged**